geode-core/src/test/java/org/apache/geode/distributed/internal/DistributionManagerDUnitTest.java

```diff
@@ -159,77 +159,74 @@ public void testConnectAfterBeingShunned() {
+            .info("<ExpectedException action=add>attempt to add old member</ExpectedException>");
         sys.getLogWriter().info(
-            "<ExpectedException action=remove>Removing shunned GemFire node</ExpectedException>");
-      }
-      mbr.setVmViewId(oldViewId);
-
-      // now forcibly add it as a surprise member and show that it is reaped
-      long gracePeriod = 5000;
-      long startTime = System.currentTimeMillis();
-      long timeout = ((GMSMembershipManager) mgr).getSurpriseMemberTimeout();
-      long birthTime = startTime - timeout + gracePeriod;
-      MembershipManagerHelper.addSurpriseMember(sys, mbr, birthTime);
-      assertTrue("Member was not a surprise member", mgr.isSurpriseMember(mbr));
-
-      // force a real view change
-      SerializableRunnable connectDisconnect = new SerializableRunnable() {
-        public void run() {
-          getSystem().disconnect();
+            "<ExpectedException action=add>Removing shunned GemFire node</ExpectedException>");
+        try {
+          boolean accepted = mgr.addSurpriseMember(mbr);
+          Assert.assertTrue("member with old ID was not rejected (bug #44566)", !accepted);
+        } finally {
+          sys.getLogWriter().info(
+              "<ExpectedException action=remove>attempt to add old member</ExpectedException>");
+          sys.getLogWriter().info(
+              "<ExpectedException action=remove>Removing shunned GemFire node</ExpectedException>");
         }
-      };
-      vm0.invoke(connectDisconnect);
-
-      if (birthTime < (System.currentTimeMillis() - timeout)) {
-        return; // machine is too busy and we didn't get enough CPU to perform more assertions
-      }
-      assertTrue("Member was incorrectly removed from surprise member set",
-          mgr.isSurpriseMember(mbr));
+        mbr.setVmViewId(oldViewId);
+
+        // now forcibly add it as a surprise member and show that it is reaped
+        long gracePeriod = 5000;
+        long startTime = System.currentTimeMillis();
+        long timeout = ((GMSMembershipManager) mgr).getSurpriseMemberTimeout();
+        long birthTime = startTime - timeout + gracePeriod;
+        MembershipManagerHelper.addSurpriseMember(sys, mbr, birthTime);
+        assertTrue("Member was not a surprise member", mgr.isSurpriseMember(mbr));
+
+        if (birthTime < (System.currentTimeMillis() - timeout)) {
+          return; // machine is too busy and we didn't get enough CPU to perform more assertions
+        }
+        assertTrue("Member was incorrectly removed from surprise member set",
+            mgr.isSurpriseMember(mbr));
 
-      try {
-        Thread.sleep(gracePeriod);
-      } catch (InterruptedException e) {
-        fail("test was interrupted", e);
-      }
+        try {
+          Thread.sleep((timeout / 3) + gracePeriod);
+        } catch (InterruptedException e) {
+          fail("test was interrupted", e);
```

Comment:
If this is a test, I recommend having "throws Exception" on the test method and then delete this try-catch block. It just adds more lines of test code and one more throwable to the stack stack.